### ISSUE TITLE
rowLink (rowModal/rowSlideover) not working in v0.6.13 / v0.7.1

### ISSUE THREAD

**Issue by J87NL** [protonemedia/laravel-splade#103]
I can't get rowLink / rowModal / rowSlideover to work, this is my setup:

```
    public function index()
    {
        return view('posts.index', [
            'dances' => SpladeTable::for(Post::class)
                ->defaultSort('title')
                ->withGlobalSearch(columns: ['title'])
                ->column(key: 'title', canBeHidden: false, sortable: true)
                ->rowLink(fn (Post $post) => route('posts.edit', ['post' => $post]))
                ->paginate(25),
        ]);
    }
```

Maybe I misread it in the documentation but the tablerows are not clickable. I tried in both Splade versions v0.6.13 and v0.7.1.
Or will this only work when I create a table class for this?

**Comment by pascalbaljet** [comment]
Yes, it should work with inline tables as well. Just checking, did you customize any of the built-in components by [publishing](https://splade.dev/docs/customization) them?

**Comment by J87NL** [comment]
Hi Pascal, thanks for your fast reply! I didn't publish the components so no changes there :+1:

**Comment by bdsumon4u** [comment]
> Hi Pascal, thanks for your fast reply! I didn't publish the components so no changes there +1

Is there any console error when you click on a row?

**Comment by bdsumon4u** [comment]
**I don't know how SpladeTable internally works.**
You've not added the `id` column to your table.
I'm not sure whether SpladeTable is selecting all columns or only specified columns from the database.
If it doesn't select the `id` column or `primary key` from the database then route-model-binding will not work.
You can try adding `id` column to your table.
`->column(key: 'id', canBeHidden: false, sortable: true)`

**Comment by J87NL** [comment]
There are console warnings when using `npm run dev` (nothing when using `npm run build`):

```
[Vue warn]: Extraneous non-props attributes (items-on-this-page, items-on-all-pages) were passed to component but could not be automatically inherited because component renders fragment or text root nodes.
```
But they are still there when I'm not adding `->rowLink()` so I don't think this is related to this issue.

Adding the id-column doesn't help but there could be something related to this causing this issue since I'm using slugs:
```
    public function getRouteKeyName(): string
    {
        return 'slug';
    }
```
Disabling this doesn't fix this issue either unfortunately.

**Comment by J87NL** [comment]
I did some more research on this myself, I'm passing in a Post:class which is an instance of Model.
In the `SpladeTable` for()-method this returns a `return new SpladeQueryBuilder($resource);` where $resource is an `Illuminate\Database\Eloquent\Builder`.

In the `SpladeQueryBuilder`'s `__construct()` this results in passing an empty resource:
`parent::__construct(resource: [], request: $request);`

I assume that's why the `HasResource` rowLink()-method has an empty `$this->resource` and fills the `$this->rowLinks` with an empty collection, resulting in not having any clickable rows in the view?

Did you manage to reproduce or is it working in your applications just fine and am I doing something wrong? 

I'm using a fresh Laravel application with laravel-splade-breeze v1.2.0 which is using laravel-splade v6.0. I tried overwriting this to v7.1 for both composer and npm but this didn't solve this issue for me.
